misk-aws/src/test/kotlin/misk/jobqueue/sqs/SqsJobQueueTest.kt

```diff
@@ -185,6 +185,62 @@ internal class SqsJobQueueTest {
     )
   }
 
+  @TestAllReceiverPolicies
+  fun setsVisibilityTimeoutOnTheMessage(receiverPolicy: String) {
+    setupReceiverPolicy(receiverPolicy)
+    val handledJobs = CopyOnWriteArrayList<Job>()
+
+    queue.enqueue(queueName, "this is my job")
+
+    val jobsReceived = AtomicInteger()
+    val allJobsCompleted = CountDownLatch(3)
+    consumer.subscribe(queueName) {
+      val sqsJob = it as SqsJob
```

Comment:
Decided to not update `Job` interface as different Misk clients could be extending it and visibilityTimeout seemed like an Sqs specific job feature.  
Open to discuss whether it would make sense to update `Job` interface.

Reply:
Maybe we can use a different name has `Job.delayForFailure()` or something like that. I feel that would be better rather than casing the real implementation.